### PR TITLE
Add ELF64 bindings

### PR DESCRIPTION
**Pull request by flysand7**
Added ELF64 bindings to mirror debug/pe package somewhat. I'll be using this to bind vdso in sys/linux package. Might also be useful for occasional users. For now limited machine-specific extensions to platforms currently supported by Odin.

### CODE REVIEW

File: examples/all/all_vendor.odin [hunk 1]
```diff
@@ -39,10 +39,10 @@ import TTF        "vendor:sdl2/ttf"
...
 import vk         "vendor:vulkan"
 
-import NS         "vendor:darwin/Foundation"
-import MTL        "vendor:darwin/Metal"
-import MTK        "vendor:darwin/MetalKit"
-import CA         "vendor:darwin/QuartzCore"
+// import NS         "vendor:darwin/Foundation"
+// import MTL        "vendor:darwin/Metal"
+// import MTK        "vendor:darwin/MetalKit"
+// import CA         "vendor:darwin/QuartzCore"
```
Comment (jakubtomsu) [inline examples/all/all_vendor.odin:45]: Did you comment this out intentionally, or was it just for testing?

Comment (flysand7) [reply examples/all/all_vendor.odin:45]: It was very intentional, but not necessary in the long run. The reason is that I was testing for compile errors locally by running:

```
$ odin run examples/all -vet -strict-style; rm all
```

And the `examples/all` package was including a bunch of darwin stuff on my linux machine which resulted in a crap ton of errors. The other PR I have pending also has these lines commented-out.

Comment (gingerBill) [reply examples/all/all_vendor.odin:45]: `examples/all` is technically not meant to use `odin build`, only `odin check`.

Comment (flysand7) [reply examples/all/all_vendor.odin:45]: Thanks for the tip!

No more accidental pushing binaries haha

**Pull request closed without merging**